src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides cannot be <= 0');
+    }
+
+    if ((a >= b + c) || (b >= a + c) || (c >= b + a)) {
+      throw new Error('The longest side should be less than sum of others');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const a = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
 
+    return Math.floor(a * 100) / 100;
```

Comment:
please move it to a separate function because you use this formula at every fugire